integrations/abrp/api.py

```diff
@@ -84,41 +103,46 @@ def __extract_basic_vehicle_status(basic_vehicle_status: BasicVehicleStatus) ->
         data = {}
 
         exterior_temperature = basic_vehicle_status.exteriorTemperature
-        if exterior_temperature is not None and exterior_temperature != -128:
+        if exterior_temperature is not None and exterior_temperature in range(-127, 127):
             data['ext_temp'] = exterior_temperature
         mileage = basic_vehicle_status.mileage
-        if mileage is not None and mileage > 0:
+        # Skip invalid range readings
+        if mileage is not None and mileage in range(1, 2147483647):
             data['odometer'] = mileage / 10.0
         range_elec = basic_vehicle_status.fuelRangeElec
-        if range_elec is not None and range_elec > 0:
+        if range_elec is not None and range_elec in range(1, 65535):
```

Comment:
Apologies if I am not understanding the code correctly. But this doesn't appear to prevent invalid range reading of 2046.0 km.